scripts/check_types.py

```diff
@@ -0,0 +1,104 @@
+"""
+Type hints checker using mypy.
```

Comment:
Could you add that this prints a tree with the current status of files wrt type checking